relay-profiling/src/transaction_metadata.rs

```diff
@@ -2,33 +2,52 @@ use serde::{Deserialize, Serialize};
 
 use relay_general::protocol::EventId;
 
-use crate::utils::deserialize_number_from_string;
+use crate::utils::{deserialize_number_from_string, is_zero};
 
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct TransactionMetadata {
     pub id: EventId,
     pub name: String,
     pub trace_id: EventId,
 
+    #[serde(
+        default,
+        deserialize_with = "deserialize_number_from_string",
+        skip_serializing_if = "is_zero"
+    )]
+    pub thread_id: u64,
```

Comment:
@phacops could we call `thread_id` `active_thread_id` instead? 

This way it's easier to figure out what it refers to.